libmamba/src/core/progress_bar.cpp

```diff
@@ -342,206 +441,1521 @@ namespace mamba
+                                fmt::fg(fmt::terminal_color::bright_cyan), "{}", "━");
+                        }
+                        else if ((i == in_progress_pos - 1) && (i < width - 1))
+                        {
+                            sstream << fmt::format(
+                                fmt::fg(fmt::terminal_color::bright_cyan), "{}", "╸");
+                        }
+                        else
+                        {
+                            sstream << fmt::format(
+                                fmt::fg(fmt::terminal_color::bright_black), "{}", "━");
+                        }
+                    }
+                    else
+                    {
+                        if (i + spinner_half_width >= pos && i <= pos + spinner_half_width)
+                        {
+                            sstream << fmt::format(
+                                fmt::fg(fmt::terminal_color::bright_magenta), "{}", "━");
+                        }
+                        else if ((pos < spinner_half_width)
+                                 && (i >= width + pos - spinner_half_width))
+                        {
+                            sstream << fmt::format(
+                                fmt::fg(fmt::terminal_color::bright_magenta), "{}", "━");
+                        }
+                        else if ((pos + spinner_half_width >= width)
+                                 && (i <= pos + spinner_half_width - width))
+                        {
+                            sstream << fmt::format(
+                                fmt::fg(fmt::terminal_color::bright_magenta), "{}", "━");
+                        }
+                        else if ((i + spinner_half_width + 1 == pos)
+                                 || ((pos < spinner_half_width)
+                                     && (i == width + pos - spinner_half_width - 1)))
+                        {
+                            sstream << fmt::format(
+                                fmt::fg(fmt::terminal_color::bright_magenta), "{}", "╺");
+                        }
+                        else if ((i == pos + spinner_half_width + 1)
+                                 || ((pos + spinner_half_width + 1 > width)
+                                     && (i == pos + spinner_half_width + 1 - width)))
+                        {
+                            sstream << fmt::format(
+                                fmt::fg(fmt::terminal_color::bright_magenta), "{}", "╸");
+                        }
+                        else
+                        {
+                            sstream << fmt::format(
+                                fmt::fg(fmt::terminal_color::bright_black), "{}", "━");
+                        }
+                    }
+                }
+            }
+        }
+
+        progress.set_value(sstream.str());
+    }
+
+    /*****************
+     * ProgressProxy *
+     *****************/
+
+    ProgressProxy::ProgressProxy(ProgressBar* ptr)
```

Comment:
I am wondering if we still need the ProgressProxy ... in fact I forgot why we need it in the first place :)

Reply:
I don't know either, the pattern was implemented by @JohanMabille  :)